packages/language-server/test/snapshots/xml-view-diagnostics/snapshots-utils.ts

```diff
@@ -14,25 +14,31 @@ export const OUTPUT_LSP_RESPONSE_FILE_NAME = "output-lsp-response.json";
 export async function snapshotTestLSPDiagnostic(
   testDir: string
 ): Promise<void> {
-  const expectedDiagnostics = getExpectedDiagnosticsLSPResponse(testDir);
-  const actualDiagnostics = await getActualDiagnosticLSPResponse(testDir);
-  expect(actualDiagnostics, "Mismatch LSP Diagnostics response").to.deep.equal(
-    expectedDiagnostics
+  const diagnosticsSnapshots = readSnapshotDiagnosticsLSPResponse(testDir);
+  const newlyComputedDiagnostics = await computeNewDiagnosticLSPResponse(
+    testDir
   );
+  expect(
+    newlyComputedDiagnostics,
+    "Snapshot Mismatch in LSP Diagnostics response"
+  ).to.deep.equal(diagnosticsSnapshots);
 
-  const expectedRanges = map(expectedDiagnostics, (_) => _.range);
-  const expectedXMLWithRangeMarkers = getExpectedXMLWithRIssueRangesMarked(
+  const newlyCommutedRanges = map(newlyComputedDiagnostics, (_) => _.range);
+  const newlyComputedXMLWithMarkedRanges = computeXMLWithMarkedRanges(
     testDir,
-    expectedRanges
+    newlyCommutedRanges
   );
-  const actualXMLWithRangeMarkers = readInputXMLSnippet(testDir, false);
+  // Note the original `input.xml` snippet acts as **both**:
+  //   - The sample input.
+  //   - A snapshot for marker ranges.
+  const originalXMLSnippet = readInputXMLSnippet(testDir, false);
   expect(
-    actualXMLWithRangeMarkers,
+    originalXMLSnippet,
     "The XML input snippet range markers are incorrect"
-  ).to.equal(expectedXMLWithRangeMarkers);
+  ).to.equal(newlyComputedXMLWithMarkedRanges);
```

Comment:
You changed this from being the snapshot ranges to the actual ranges, which changes the purpose of this assertion - instead of making sure the input xml and expected response json are in sync, it checks that the ranges in the actual response are as expected in the input xml, I assume for better error messages. But if they aren't (and assuming the input xml and resopnse json are in sync) it will already fail in line 24...

Reply:
I was trying to resolve the confusion we had yesterday due to mixing up the input and the snapshot.

So I am treating them as two **distinct** unrelated snapshots. Which are tested separately and visually inspected seperatly.

The assumption: `the input xml and response json are in sync` is not a safe assumption as the input.xml can be edited by **human** developers, it is not necessarily auto-generated.

Reply:
Could you add a check for the consistency of the ranges in the input xml and snapshot json? Preferably at the beginning of the function. It would make it easier to understand if a test fails due to a bug in the code or because there's a problem in the xml/snapshot json.
It would also make it easier to read the tests if you assume the 2 files are in sync. No need to figure out the ranges in the json.

Reply:
I don't understand the benefits, lets sync in person on this (maybe tomorrow?)

Reply:
ok